File: Triatlon-Java/TriatlonPersistance/src/main/java/repository/databases/ArbitruDB.java
package repository.databases;

import domain.*;
import repository.Utils.JdbcUtils;
import repository.interfaces.ArbitruRepository;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.io.FileNotFoundException;
import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class ArbitruDB implements ArbitruRepository {

    private final JdbcUtils jdbcUtils;

    private static final Logger logger = LogManager.getLogger();

    public ArbitruDB(Properties props) {
        logger.info("Initializing ArbitruDB with properties: {} ", props);
        System.out.println("Initializing ArbitruDB with properties: {} "+ props);
        jdbcUtils = new JdbcUtils(props);
    }


    @Override
    public Arbitru findOne(Long aLong) {
        logger.traceEntry("finding arbitru with {}", aLong);
        Connection con = jdbcUtils.getConnection();
        try(PreparedStatement preStmt = con.prepareStatement("select * from arbitru where id= ?")) {
            preStmt.setLong(1, aLong);
            try(ResultSet result = preStmt.executeQuery()) {
                if(result.next()) {
                    String nume = result.getString("nume");
                    String prenume = result.getString("prenume");
                    String username = result.getString("username");
                    String parola = result.getString("parola");
                    Arbitru arbitru = new Arbitru(nume, prenume, username, parola);
                    arbitru.setId(aLong);
                    logger.traceExit(arbitru);
                    return arbitru;
                }
            } catch (SQLException e) {
                logger.error(e);
                System.err.println("Error DB " + e);
            }
        } catch (SQLException e) {
            logger.error(e);
            System.err.println("Error DB " + e);
        }
        return null;
    }

    @Override
    public Iterable<Arbitru> findAll() throws SQLException {
        logger.traceEntry();
        Connection con = jdbcUtils.getConnection();
        List<Arbitru> arbitri = new ArrayList<>();
        try(PreparedStatement preStmt = con.prepareStatement("select * from arbitru")) {
            try(ResultSet result = preStmt.executeQuery()) {
                while(result.next()) {
                    Long id = result.getLong("id");
                    String nume = result.getString("nume");
                    String prenume = result.getString("prenume");
                    String username = result.getString("username");
                    String parola = result.getString("parola");
                    Arbitru arbitru = new Arbitru(nume, prenume,username,  parola);
                    arbitru.setId(id);
                    arbitri.add(arbitru);
                    logger.traceExit(arbitri);
                }

            } catch (SQLException e) {
                logger.error(e);
                System.err.println("Error DB " + e);
            }
            return arbitri;
        }
    }

    @Override
    public Arbitru save(Arbitru entity) throws FileNotFoundException {
        logger.traceEntry("saving arbitru {} ", entity);
        Connection con = jdbcUtils.getConnection();
        try(PreparedStatement preStmt = con.prepareStatement("insert into arbitru (nume, prenume, username, parola) values (?, ?, ?, ?)")) {
            preStmt.setString(1, entity.getNume());
            preStmt.setString(2, entity.getPrenume());
            preStmt.setString(3, entity.getUsername());
            preStmt.setString(3, entity.getParola());
            int result = preStmt.executeUpdate();
            logger.trace("Saved {} instances", result);
        } catch (SQLException e) {
            logger.error(e);
            System.err.println("Error DB " + e);
        }
        logger.traceExit();
        return null;
    }

    @Override
    public Arbitru delete(Long aLong) {
        return null;
    }

    @Override
    public Arbitru update(Arbitru entity1, Arbitru entity2) {
        return null;
    }

    @Override
    public Arbitru getAccount(String username, String parola) {
        logger.traceEntry("finding arbitru with {}", username);
        Connection con = jdbcUtils.getConnection();
        try(PreparedStatement preStmt = con.prepareStatement("select * from arbitru where username = ? and parola = ?")) {
            preStmt.setString(1, username);
            preStmt.setString(2, parola);
            try(ResultSet result = preStmt.executeQuery()) {
                if(result.next()) {
                    Long id = result.getLong("id");
                    String nume = result.getString("nume");
                    String prenume = result.getString("prenume");
                    Arbitru arbitru = new Arbitru(nume, prenume, username, parola);
                    arbitru.setId(id);
                    logger.traceExit(arbitru);
                    return arbitru;
                }
            } catch (SQLException e) {
                logger.error(e);
                System.err.println("Error DB " + e);
            }
        } catch (SQLException e) {
            logger.error(e);
            System.err.println("Error DB " + e);
        }
        return null;
    }
}

File: Triatlon-C#/TriatlonPersistance/databases/ArbitruDB.cs
using System;
using System.Collections.Generic;
using log4net;
using System.Data.SQLite;
using System.Windows.Forms;
using Triatlon.repository.interfaces;
using TriatlonModel;

namespace ConcursTriatlon.repository.databases
{
    public class ArbitruDB : ArbitruRepository
    {
        private static readonly ILog log = LogManager.GetLogger("ArbitruDB");
        private string connectionString;

        public ArbitruDB(string connectionString)
        {
            log.Info("Creating ArbitruDB");
            this.connectionString = connectionString;
        }

        public Arbitru getAccount(string username, string parola)
        {
            log.InfoFormat("Entering getAccount with value {0}", username, parola);
            using (SQLiteConnection connection = new SQLiteConnection(connectionString))
            {
                try
                {
                    connection.Open();
                    SQLiteCommand command = new SQLiteCommand("select * from arbitru where username = @username and parola = @parola", connection);
                    command.Parameters.AddWithValue("@username", username);
                    command.Parameters.AddWithValue("@parola", parola);
                    SQLiteDataReader reader = command.ExecuteReader();
                    while (reader.Read())
                    {
                        long id = reader.GetInt64(0);
                        string nume = reader.GetString(1);
                        string prenume = reader.GetString(2);

                        Arbitru arbitru = new Arbitru(nume, prenume, parola, username);
                        arbitru.setId(id);
                        reader.Close();

                        return arbitru;
                    }
                }
                catch (Exception e)
                {
                    log.Error(e.Message);
                    MessageBox.Show(e.Message);
                }
            }
            log.InfoFormat("Exiting getAccount with value {0}", null);

            return null;
        }

        public Arbitru findOne(long id)
        {
            log.InfoFormat("Entering findOne with value {0}", id);
            using (SQLiteConnection connection = new SQLiteConnection(connectionString))
            {
                try
                {
                    connection.Open();
                    SQLiteCommand command = new SQLiteCommand("select * from arbitru where id=@id", connection);
                    command.Parameters.AddWithValue("@id", id);
                    SQLiteDataReader reader = command.ExecuteReader();
                    while (reader.Read())
                    {
                        string nume = reader.GetString(1);
                        string prenume = reader.GetString(2);
                        string parola = reader.GetString(3);
                        string username = reader.GetString(4);
                        Arbitru arbitru = new Arbitru(nume, prenume, parola,username);
                        arbitru.setId(id);
                        reader.Close();

                        return arbitru;
                    }
                } catch (Exception e)
                {
                    log.Error(e.Message);
                    MessageBox.Show(e.Message);
                }
                
            }
            log.InfoFormat("Exiting findOne with value {0}", null);
            return null;
        }

        public IEnumerable<Arbitru> findAll()
        {
            log.Info("Entering findAll");
            List<Arbitru> arbitri = new List<Arbitru>();
            using (SQLiteConnection connection = new SQLiteConnection(connectionString))
            {
                try
                {
                    connection.Open();
                    SQLiteCommand command = new SQLiteCommand("select * from arbitru", connection);
                    SQLiteDataReader reader = command.ExecuteReader();
                    while (reader.Read())
                    {
                        long id = reader.GetInt64(0);
                        string nume = reader.GetString(1);
                        string prenume = reader.GetString(2);
                        string parola = reader.GetString(3);
                        string username = reader.GetString(4);

                        Arbitru arbitru = new Arbitru(nume, prenume, parola, username);
                        arbitru.setId(id);
                        arbitri.Add(arbitru);
                    }
                    reader.Close();

                }
                catch (Exception e)
                {
                    log.Error(e.Message);
                    MessageBox.Show(e.Message);
                }
            }
            log.InfoFormat("Exiting findAll with value {0}", arbitri);
            return arbitri;
        }

        public Arbitru save(Arbitru entity)
        {
            //Console.WriteLine("save");
            log.InfoFormat("Entering save with value {0}", entity);
            using (SQLiteConnection connection = new SQLiteConnection(connectionString))
            {
                try
                {
                    connection.Open();
                    SQLiteCommand command = new SQLiteCommand("insert into arbitru(nume, prenume, parola,username) values (@nume, @prenume, @parola,@username)", connection);
                    command.Parameters.AddWithValue("@nume", entity.Nume);
                    command.Parameters.AddWithValue("@prenume", entity.Prenume);
                    command.Parameters.AddWithValue("@parola", entity.Parola);
                    command.Parameters.AddWithValue("@username", entity.Username);
                    command.ExecuteNonQuery();
                } catch (Exception e)
                {   
                    log.Error(e.Message);
                    MessageBox.Show(e.Message);
                }
            }
            log.InfoFormat("Exiting save with value {0}", null);
           
            return null;
        }

        public Arbitru delete(long id)
        {
            return null;
        }

        public Arbitru update(Arbitru entity1, Arbitru entity2)
        {
            return null;
        }

        
    }
}